src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -529,6 +568,97 @@ private CachingAmazonDynamoDbStreams(AmazonDynamoDBStreams amazonDynamoDbStreams
         GuavaCacheMetrics.monitor(meterRegistry, iteratorCache, cn + ".GetShardIterator");
     }
 
+    /**
+     * Gets the {@code DescribeStreamResult} from cache or from DescribeStream API.
+     * @param describeStreamRequest Describe stream request.
+     * @return Stream details for the given request (streamArn).
+     */
+    private DescribeStreamResult lookupDescribeStreamResult(DescribeStreamRequest describeStreamRequest)
+        throws AmazonDynamoDBException {
+
+        String key = describeStreamRequest.getStreamArn();
+        try {
+            DescribeStreamResult result = describeStreamCache.get(key,
+                () -> this.loadStreamDescriptionForAllShards(describeStreamRequest));
```

Comment:
Would it make sense to configure the loading method in when creating the cache, similar to what we do for the iterator cache?

Reply:
For consistency?